src/librustc_parse/parser/expr.rs

```diff
@@ -195,6 +195,28 @@ impl<'a> Parser<'a> {
                     return Ok(expr);
                 }
             }
+
+            if (op.node == AssocOp::Equal || op.node == AssocOp::NotEqual)
+                && self.token.kind == token::Eq
```

Comment:
how does this deal with `a == = B`?

Reply:
It will suggest it's removal. It doesn't differentiate between `===` and `== =`. Didn't add any logic to do so because the later would never be valid logic. Similarly, this won't trigger on `= ==`, which I thought acceptable.

Reply:
My main concern is that the error message will mention `===` which isn't what's actually written, but that really isn't important enough to spend a lot of time on :laughing: so feel free to go ahead and merge this even if you don't want to fix this

Reply:
Done :)